index.js

```diff
@@ -16,15 +21,61 @@ async function run() {
     const logConfigurationOptions = core.getInput("log-configuration-options", { required: false });
     const dockerLabels = core.getInput('docker-labels', { required: false });
     const command = core.getInput('command', { required: false });
+    
+     //New inputs to fetch task definition 
+     const taskDefinitionArn = core.getInput('task-definition-arn', { required: false });
+     const taskDefinitionFamily = core.getInput('task-definition-family', { required: false });
+     const taskDefinitionRevision = Number(core.getInput('task-definition-revision', { required: false}));
+
+    let taskDefPath;
+    let taskDefContents;
+    let describeTaskDefResponse;
+
+    if(taskDefinition){
+      core.warning("Task definition file will be used.");
+      taskDefPath = path.isAbsolute(taskDefinition) ?
+      taskDefinition : 
+      path.join(process.env.GITHUB_WORKSPACE, taskDefinition); 
+      if (!fs.existsSync(taskDefPath)) {
+        throw new Error(`Task definition file does not exist: ${taskDefinition}`);
+      }
+      taskDefContents = require(taskDefPath);
+    }
+
+    else if(taskDefinitionArn || taskDefinitionFamily || taskDefinitionRevision){
+      if(taskDefinitionArn){
+        core.warning("The task definition arn will be used to fetch task definition");
+      }
+      if(taskDefinitionFamily && taskDefinitionRevision && !taskDefinitionArn){
```

Comment:
Nit: It would be more efficient to replace/re-write inner if statements as if/else-if blocks.